### PR TITLE
Lint sublists in `list-item`

### LINKED ISSUE #57: Sub-lists are not checked for the `list-item` rule
This readme passes `awesome-lint`: https://github.com/juliandavidmr/awesome-nestjs/blob/a0c49d8dc6afa67a0c64f332fd59daccfe5b208f/README.md#resources But a lot of the descriptions are missing a `.` at the end.

We should ensure the other custom rules work with nested list items too.

// @transitive-bullshit

### PR DESCRIPTION
**Pull request by itaisteinherz**
Finally got to fixing #57 😅

Fixes #57

### CODE REVIEW

File: rules/list-item.js [hunk 1]
```diff
@@ -81,6 +81,36 @@ function findAllLists(ast) {
 	return lists;
 }
 
+function findSublists(node) {
```
Comment (transitive-bullshit) [inline rules/list-item.js]: I'd prefer to use https://github.com/syntax-tree/unist-util-visit instead of rewriting the recursive visitor pattern here.

Comment (itaisteinherz) [reply rules/list-item.js]: For some reason this didn't work at first, but now it works as expected 🤷🏻‍♂️

**Comment by itaisteinherz** [comment]
@transitive-bullshit Ready to merge ✅

**Comment by transitive-bullshit** [comment]
Thanks Itai!!

**Pull request merged**